File: client-clojure/src/sirqul_io_t_platform/specs/step_response.clj
(ns sirqul-io-t-platform.specs.step-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.leg-response :refer :all]
            )
  (:import (java.io File)))


(def step-response-data
  {
   (ds/opt :leg) leg-response-spec
   (ds/opt :distance) float?
   (ds/opt :duration) int?
   (ds/opt :startLat) float?
   (ds/opt :startLng) float?
   (ds/opt :startAlt) float?
   (ds/opt :startDate) int?
   (ds/opt :endLat) float?
   (ds/opt :endLng) float?
   (ds/opt :endAlt) float?
   (ds/opt :endDate) int?
   (ds/opt :accuracy) int?
   })

(def step-response-spec
  (ds/spec
    {:name ::step-response
     :spec step-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/StepResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import StepResponse._

case class StepResponse (
  leg: Option[LegResponse],
distance: Option[Double],
duration: Option[Long],
startLat: Option[Double],
startLng: Option[Double],
startAlt: Option[Double],
startDate: Option[Long],
endLat: Option[Double],
endLng: Option[Double],
endAlt: Option[Double],
endDate: Option[Long],
accuracy: Option[Integer])

object StepResponse {
  import DateTimeCodecs._

  implicit val StepResponseCodecJson: CodecJson[StepResponse] = CodecJson.derive[StepResponse]
  implicit val StepResponseDecoder: EntityDecoder[StepResponse] = jsonOf[StepResponse]
  implicit val StepResponseEncoder: EntityEncoder[StepResponse] = jsonEncoderOf[StepResponse]
}
